File: 433.c
#include <stdio.h>

int sgcd(int x, int y)
{
    int cnt = 0;
    int t;
    while(1){
        t = x;
        x = y;
        y = t % y;
        cnt++;
        if(y == 0)
            return cnt;
    }
}


int Sgcd(int n)
{
    int cnt = 0,i,j;
    for(i = 1; i <= n; i++){
        printf("\r%16d %16d",i,cnt);
        cnt ++;
        for(j = 1; j < i; j++){
            cnt += 2 * sgcd(i, j) + 1;
        }
    }
    return cnt;
}

void main(){
    printf("\n");
    printf("\n%d\n",Sgcd(1000*1000));
}

File: 433.py
def sgcd(x, y):
    cnt = 0
    while 1:
        x, y = y, x % y
        cnt += 1
        if y == 0:
            return cnt


def ssgcd(n):
    ss = n;
    for x in xrange(1, n + 1):
        print "\r%16d %16d" % (x, ss),
        for y in xrange(1, x):
            ss += 2 * sgcd(x, y) + 1
    return ss


for i in range(1, 40 + 1):
    for j in range(1, i):
        print "%d " % (sgcd(i,j)),
    print "*"
